crates/bevy_ecs/src/query/iter.rs

```diff
@@ -404,19 +352,20 @@ impl<'w, 's, Q: WorldQuery> ExactSizeIterator for QueryIter<'w, 's, Q, ()> {
     }
 }
 
-struct QueryIterationCursor<'s, Q: WorldQuery, F: WorldQuery> {
+struct QueryIterationCursor<'w, 's, Q: WorldQuery, QF: Fetch<'w, State = Q::State>, F: WorldQuery> {
     table_id_iter: std::slice::Iter<'s, TableId>,
     archetype_id_iter: std::slice::Iter<'s, ArchetypeId>,
-    fetch: Q::Fetch,
+    fetch: QF,
     filter: F::Fetch,
     current_len: usize,
     current_index: usize,
     is_dense: bool,
+    phantom: PhantomData<&'w Q>,
```

Comment:
I believe that `QF: Fetch<'s>` and dropping the `Q` and `'w` would work for this, as I believe the lifetime parameter on `Fetch` is "lasts at least as long" but I wasn't 100% sure so went with a more obviously "correct" solution of including them with a PhantomData.